experiment/seccomp/src/instruction/arch.rs

```diff
@@ -0,0 +1,20 @@
+use crate::instruction::Instruction;
+use crate::instruction::*;
+
+pub enum Arch {
+    X86,
+}
+
+impl Arch {
+    pub fn gen_validate(&self) -> Vec<Instruction> {
```

Comment:
For this, can we make this as an interface? For example, something like we have done in cgroups or syscall, so we have a function that will return dyn Arch, and we call method on it.

Alternatively, if Arch itself does not need to hold any data, its just an enum, can we instead have an empty struct, and have static methods (no &self) , and inside those methods, we either match over the arch at runtime, or do conditional compilation based on cfg. That way adding support for new arch will be easier.

Reply:
I prefer later 👍
3266094a